lib/rpc/src/rpc_api.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright © 2021 University of Colorado. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use alloc::vec::Vec;
+use core::result::Result;
+
+use crate::cluster_api::NodeId;
+use crate::rpc::{RPCError, RPCHeader, RPCType};
+
+/// RPC server operations
+pub trait RPCServerAPI {
+    /// register an RPC func with an ID
+    fn rpc_register(&self, rpc_id: RPCType) -> Result<(), RPCError>;
```

Comment:
I'd drop the `rpc_*` prefix for naming here as it is inherent in the type/nameof the struct